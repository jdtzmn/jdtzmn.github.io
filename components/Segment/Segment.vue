<template lang="pug">
  .segment
    .segment-container
      slot
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.component('segment', {})
</script>

<style lang="scss" scoped>
@import '~@/styles/grid';
@import '~@/styles/breakpoints';
@import '~@/styles/font';

.segment-container {
  // display
  @extend %center-block;
  padding: 50px 25px;

  @include lg {
    padding-top: 75px;
    padding-bottom: 75px;
  }
}

.segment-container > .segment-header {
  @extend %header1;

  // display
  padding-bottom: 15px;

  // font
  font-size: 1.8em;
  font-weight: 500;
}

// =========== STYLES ===========

// slanted
.segment.slanted {
  // transform
  transform: skewY(2deg);

  & > .segment-container {
    // transform
    transform: skewY(-2deg);
  }
}

</style>
