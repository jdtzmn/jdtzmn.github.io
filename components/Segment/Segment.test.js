import { shallowMount } from '@vue/test-utils'
import Segment from './Segment'

describe('Segment', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Segment, {
      slots: {
        default: '<div class="slot"></div>'
      }
    })
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render segment div', () => {
    expect(wrapper.contains('.segment')).toBeTruthy()
  })

  it('should render segment-content div inside segment div', () => {
    expect(wrapper
      .find('.segment')
      .contains('.segment-container'))
      .toBeTruthy()
  })

  it('slot should render inside segment div', () => {
    expect(wrapper
      .find('.segment')
      .find('.segment-container')
      .contains('.slot'))
      .toBeTruthy()
  })
})
