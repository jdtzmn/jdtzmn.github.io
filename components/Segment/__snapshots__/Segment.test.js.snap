// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segment should match snapshot 1`] = `
<div class="segment">
  <div class="segment-container">
    <div class="slot"></div>
  </div>
</div>
`;
